### PR TITLE
[Finishes #101814312] image('id', ''); shouldn't fetch from Assets API

### PR DESCRIPTION
**Pull request by joshlory**
This is actually a bug in the short-url rewrite. An image with `src=""` is correctly hidden, but what we're actually putting in the DOM is ``.

### CODE REVIEW

**Comment by davidsbailey** [comment]
Apparently this doesn't quite solve the problem as it may still hit various parts of the site: 

http://stackoverflow.com/questions/19126185/setting-an-image-src-to-empty
https://www.nczonline.net/blog/2009/11/30/empty-image-src-can-destroy-your-site/

If you don't like the inline data image, we could return "/blockly/media/1x1.gif" instead.

**Comment by davidsbailey** [comment]
possibly useful example project for browser testing with two empty image sources: https://studio.code.org/projects/applab/oisQ8J8oDRFL_Rfo8GppdQ

**Comment by davidsbailey** [comment]
this should go without saying, but let me know if you disagree and we can discuss

**Comment by joshlory** [comment]
@davidsbailey ready for another look.

**Comment by davidsbailey** [comment]
LGTM

File: apps/src/applab/applab.js [hunk 1]
```diff
@@ -1519,6 +1519,10 @@ Applab.maybeAddAssetPathPrefix = function (filename) {
   }
 
   filename = filename || '';
+  if (filename.length === 0) {
+    return '/blockly/media/1x1.gif';
+  }
```
Comment (Bjvanminnen) [inline apps/src/applab/applab.js:1524]: could we have also done `filename = filename || '/blockly/media/1x1.gif'?`

**Pull request merged**